### PR TITLE
Edit form

### LINKED ISSUE #6: [F] Edit past connection
Scenario: A database has already been added and needs to be modified. 
Given the user has an existing database that they have connected to before; 
When the user chooses the existing configuration and the modify button is pressed; 
Then the user will be able to change the previously entered connection settings. 

AC:
- On Recent Connections page, each connection should have a "modify" button (pencil?)
- Send new information to backend

### PR DESCRIPTION
**Pull request by Smeanie**
# Description

Add an edit option to previous connections and a form to change 

## Link to the issue you are closing

Fixes #6 

## Screenshot or other testing that you have completed
image

Your PR should include at least one test/screenshot in order to showcase that the changes don't break/fix what was needed. In the case of frontend, a screenshot should suffice. For backend, depending on the scenario, some temporary tests may need to be written to show that the function does what's needed.

# Checklist:

- [x] ESLint passes
- [x] Prettier passes
- [x] Included comments where necessary
- [x] Updated README if needed
- [x] Verified that code runs and generates no errors/warnings

### CODE REVIEW

**Review by adityavidyadharan — CHANGES_REQUESTED** [review]
The edit form was not working for me when I pulled. The only field I could edit was the nickname field and for the changes to show you have to go to the home page then back to the recents page. All of the other fields would not save.

We don't need the ability to edit all the fields and error check them, **we only need to edit the nickname** for the demo

Also, the delete button uses an icon on `feature`. You can pull the branch to see the library I used. It also has pencil icons that can replace the "Edit" button.

File: src/renderer/scss/RecentConnections.scss [hunk 1]
```diff
@@ -53,3 +53,31 @@ td {
...
+
+.edit-button:hover {
+  background-color: lightgreen;
+}
+
+.edit-submit-btn {
+  margin: 10px 0px;
+}
+
+.edit-button {
```
Comment (alex-hobby) [inline src/renderer/scss/RecentConnections.scss:71]: Not sure how these classes got in there twice - let's remove the duplicates

**Review by alex-hobby — CHANGES_REQUESTED** [review]

File: src/renderer/components/RecentConnections.tsx [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 import '../scss/RecentConnections.scss';
 
 const RecentConnections = () => {
   const [connect, setConnect] = useState<ConnectionModel[]>([]);
+  const getConnections = async () => {
+    const connections = await window.connections.ipcRenderer.fetch();
+    setConnect(connections);
+  };
   useEffect(() => {
-    const getConnections = async () => {
```
Comment (alex-hobby) [inline src/renderer/components/RecentConnections.tsx:13]: this change is unnecessary - getConnections only needs to be in scope for the useEffect

Comment (alex-hobby) [reply src/renderer/components/RecentConnections.tsx:13]: never mind, I missed that it was used below

Comment (insaan21) [reply src/renderer/components/RecentConnections.tsx:13]: We can remove the declaration that's in scope though I thought I had done that in my PR but guess not

Comment (insaan21) [reply src/renderer/components/RecentConnections.tsx:13]: Oh wait im trolling lol

Comment (alex-hobby) [reply src/renderer/components/RecentConnections.tsx:13]: right, is this useEffect even needed?

Comment (insaan21) [reply src/renderer/components/RecentConnections.tsx:13]: Yeah the call to getConnections() needs to be inside a useEffect()

File: src/renderer/scss/RecentConnections.scss (hunk 1 above)
Comment (insaan21) [reply src/renderer/scss/RecentConnections.scss]: Removed

**Review by insaan21 — APPROVED** [review]
wow

**Review by alex-hobby — APPROVED** [review]
👍

**Pull request merged**